### ISSUE TITLE
Update asa.c to safely remain on CRAN

### ISSUE THREAD

**Issue by mhunter1** [OpenMx/OpenMx#394]
> Please see the problems shown on
> 
> <https://cran.r-project.org/web/checks/check_results_OpenMx.html>.
> 
> Please correct before 2024-08-20 to safely retain your package on CRAN.

It looks like a new set of warnings on Linux under the dev version of R.  They're coming from the asa.c file which does Adaptive Simulated Annealing.  That file does something that R no longer likes: it uses #if and #else statements inside a call to fprintf.  One is copied below.

The fix is well within my capabilities.  I'm happy to fix this if no one else is itching to do so.

```
fprintf (ptr_asa_out,
#if INT_LONG
                 "best...->cost=%-*.*g  *number_accepted=%ld  *number_generated=%ld\n",
                 G_FIELD, G_PRECISION, best_generated_state->cost,
#else
                 "best...->cost=%-*.*g  *number_accepted=%d  *number_generated=%d\n",
                 G_FIELD, G_PRECISION, best_generated_state->cost,
#endif /* INT_LONG */
                 *number_accepted, *number_generated);
```

**Comment by mhunter1** [comment]
-    [ ]     asa.c:981:2: warning: embedding a directive within macro arguments has undefined behavior [-Wembedded-directive]
-    [ ]     asa.c:984:2: warning: embedding a directive within macro arguments has undefined behavior [-Wembedded-directive]
-    [ ]     asa.c:986:2: warning: embedding a directive within macro arguments has undefined behavior [-Wembedded-directive]
-    [ ]     asa.c:989:2: warning: embedding a directive within macro arguments has undefined behavior [-Wembedded-directive]
-    [ ]     asa.c:1020:2: warning: embedding a directive within macro arguments has undefined behavior [-Wembedded-directive]
-    [ ]     asa.c:1023:2: warning: embedding a directive within macro arguments has undefined behavior [-Wembedded-directive]
-    [ ]     asa.c:1025:2: warning: embedding a directive within macro arguments has undefined behavior [-Wembedded-directive]
-    [ ]     asa.c:1028:2: warning: embedding a directive within macro arguments has undefined behavior [-Wembedded-directive]
-    [ ]     asa.c:2785:2: warning: embedding a directive within macro arguments has undefined behavior [-Wembedded-directive]
-    [ ]     asa.c:2788:2: warning: embedding a directive within macro arguments has undefined behavior [-Wembedded-directive]
-    [ ]     asa.c:3767:2: warning: embedding a directive within macro arguments has undefined behavior [-Wembedded-directive]
-    [ ]     asa.c:3770:2: warning: embedding a directive within macro arguments has undefined behavior [-Wembedded-directive]
-    [ ]     asa.c:3772:2: warning: embedding a directive within macro arguments has undefined behavior [-Wembedded-directive]
-    [ ]     asa.c:3775:2: warning: embedding a directive within macro arguments has undefined behavior [-Wembedded-directive]
-    [ ]     asa.c:6771:2: warning: embedding a directive within macro arguments has undefined behavior [-Wembedded-directive]
-    [ ]     asa.c:6774:2: warning: embedding a directive within macro arguments has undefined behavior [-Wembedded-directive]
-    [ ]     asa.c:6776:2: warning: embedding a directive within macro arguments has undefined behavior [-Wembedded-directive]
-    [ ]     asa.c:6779:2: warning: embedding a directive within macro arguments has undefined behavior [-Wembedded-directive]
-    [ ]     asa.c:6806:2: warning: embedding a directive within macro arguments has undefined behavior [-Wembedded-directive]
-    [ ]     asa.c:6809:2: warning: embedding a directive within macro arguments has undefined behavior [-Wembedded-directive]
-    [ ]     asa.c:6811:2: warning: embedding a directive within macro arguments has undefined behavior [-Wembedded-directive]
-    [ ]     asa.c:6814:2: warning: embedding a directive within macro arguments has undefined behavior [-Wembedded-directive]
-    [ ]     asa.c:6819:2: warning: embedding a directive within macro arguments has undefined behavior [-Wembedded-directive]
-    [ ]     asa.c:6822:2: warning: embedding a directive within macro arguments has undefined behavior [-Wembedded-directive]
-    [ ]     asa.c:6824:2: warning: embedding a directive within macro arguments has undefined behavior [-Wembedded-directive]
-    [ ]     asa.c:6827:2: warning: embedding a directive within macro arguments has undefined behavior [-Wembedded-directive]

**Comment by mhunter1** [comment]
Closed by 7462f5758d5eb638a782b28dccb8db5dd5aadc0d
